.github/workflows/ci.yml

```diff
@@ -0,0 +1,54 @@
+name: Enr CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+env:
+  ZIG_VERSION: 0.14.1
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4  
+
+      - name: Cache Zig
+        id: cache-zig  
+        uses: actions/cache@v4  
+        with:
+          path: ~/zig
+          key: ${{ runner.os }}-zig-${{ env.ZIG_VERSION }}
+
+      - name: Install Zig
+        if: steps.cache-zig.outputs.cache-hit != 'true'
+        run: |
+          wget https://ziglang.org/download/${{ env.ZIG_VERSION }}/zig-x86_64-linux-${{ env.ZIG_VERSION }}.tar.xz
+          tar -xf zig-x86_64-linux-${{ env.ZIG_VERSION }}.tar.xz
+          mv zig-x86_64-linux-${{ env.ZIG_VERSION }} ~/zig
```

Comment:
Look at the main zeam repo so see how we do that, use an action instead of implementing it yourself.